sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -194,10 +195,17 @@ case class FileSourceScanExec(
     disableBucketedScan: Boolean = false)
   extends DataSourceScanExec {
 
+  lazy val metadataStructCol: Option[AttributeReference] =
```

Comment:
Are we assuming that only 1 column in `output` corresponds to `MetadataAttribute(_)`? Is there some place in code where we are enforcing this?